lib/galaxy/config/__init__.py

```diff
@@ -268,19 +268,28 @@ def _in_dir(self, _dir, path):
         return os.path.join(_dir, path) if path else None
 
     def _parse_config_file_options(self, defaults, listify_defaults, config_kwargs):
+
+        def root_join(val):
+            try:
+                return os.path.join(self.root, val)
+            except TypeError:
+                template = "Failed to set configuration variable %s, value %s of wrong type %s"
+                message = template % (var, val, type(val))
+                raise Exception(message)
```

Comment:
if we were to raise an exception, I would suggest a `ConfigurationError` like here https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/config/__init__.py#L834

Reply:
```suggestion
                raise ConfigurationError(message)
```